### PR TITLE
feat(database): sqlite plugin rollback support

### PR DESCRIPTION
**Pull request by wolf31o2**
## Summary by cubic
Adds rollback support to the SQLite metadata plugin to safely restore state after chain rollbacks. Handles certificates, protocol params, and account/pool/DRep state with deterministic ordering and strong FK enforcement.

- **New Features**
  - Rollback APIs to delete records after a given slot: certificates, protocol params, and param updates.
  - Deterministic restoration using slot/blockIndex/certIndex ordering; batch caches for accounts, pools, and DReps to avoid N+1 queries.
  - Batch UTXO fetch (GetUtxosBatch) with safe chunking under SQLite bind limits.
  - Large test suite covering rollbacks and edge cases.

- **Refactors**
  - Enforce SQLite foreign keys via DSN (_pragma=foreign_keys(1)); add sqliteBindVarLimit for safe IN clauses.
  - Race-safe account creation pattern; clearer errors/logging.

<sup>Written for commit 132a89fad6f02a0ba331ec8cbe322a91aaac722c. Summary will update on new commits.</sup>

### CODE REVIEW

File: database/plugin/metadata/sqlite/database.go [hunk 1]
```diff
@@ -228,7 +233,7 @@ func (d *MetadataStoreSqlite) Start() error {
 		//
 		// Keep these parameters in sync with project performance testing.
 		connStr := fmt.Sprintf(
-			"file:%s?cache=shared&_journal_mode=WAL&sync=OFF&cache_size=-50000",
+			"file:%s?cache=shared&_journal_mode=WAL&sync=NORMAL&cache_size=-50000&_pragma=foreign_keys(1)",
```
Comment (agaffney) [inline database/plugin/metadata/sqlite/database.go]: Setting `sync=NORMAL` is going to cause a performance hit
